### ISSUE TITLE
Mining pool compatibility

### ISSUE THREAD

**Issue by jujumax** [tothemoon-org/extension-blocks#13]
Extension blocks require to include the merkle root of the extblk into the coinbase TX in the canonical block also the last TX needs to be the resolution TX. Today mining pool software (NOMP, CKPool. eloipool) create the coinbase TX and also the merkle root and TX sort into the tree. To deploy extblocks mining pool software needs to be updated, a reference implementation will be nice.

**Comment by chjj** [comment]
Yeah, this is true. It's primarily the commitment output that's the issue. If no one else does it, I may fork ckpool or eloipool to get something working.

**Comment by jujumax** [comment]
Probably NOMP is simpler for you to modify because is node.js, we can work in ckpool fork.

**Comment by jujumax** [comment]
If we left the coinbase TX outputs as is and use the coinbase message to reference to a TX number where with a nonstandard TX we include the merkle of the extblk we use an extra TX but we do not have to modify the mining software at all and will help adoption by pools.
